packages/@react-aria/table/src/TableKeyboardDelegate.ts

```diff
@@ -149,42 +152,45 @@ export class TableKeyboardDelegate<T> implements KeyboardDelegate {
     // If no item was found, and focus was on a cell, then focus the
     // corresponding column header.
     if (this.isCell(startItem)) {
-      return this.collection.columns[startItem.index].key;
+      // If current selectionMode is 'single', focus should not be moved to the missing "select all" checkbox column, instead move focus to the 2nd column
+      let index = startItem.index === 0 && this.selectionMode === 'single' ? startItem.index + 1 : startItem.index;
```

Comment:
Many of the changes here are to stop focus from moving to the first column header div when `selectionMode = 'single'` since the "Select All" checkbox isn't rendered in that case. Ideally it would be nice not to render that div at all, but stuff breaks (need to dig more)

Reply:
if we disable that cell, will it get skipped automagically?

Reply:
Unfortunately it doesn't get skipped with isDisabled only. From what I can tell, the collection's keyMap doesn't have a notion of disabled items. The items retrieved by `collection.getItem` also don't contain any information about their own `disabled` state that I can find. Right now disabled rows are skipped over because they are contained within the `disabledKeys` prop which the delegate has access to

It _might_ be possible to change CollectionBuilder so it attaches that info somehow, but I'm not sure if that is feasible